File: 99-old-code/7.SpringMvc/in28minutes-services-springmvc/src/main/webapp/resources/js/application/controllers/TodoController.js
var TodoController = function($scope, $http){

    $scope.editMode = false;
    $scope.position = '';

    $scope.getAllTodos = function(){
        $scope.resetError();
        $http.get('todo/all.json').success(function(response){
            $scope.todos = response;
        }).error(function() {
            $scope.setError('Could not display all todos');
        });
    }

    $scope.addTodo = function(newTodo){
        $scope.resetError();
        $http.post('todo/add/' + newTodo).success(function(response){
            $scope.getAllTodos();
        }).error(function() {
            $scope.setError('Could add todo');
        });
        $scope.todoName = '';
    }

    $scope.deleteTodo = function(deleteTodo){
        $scope.resetError();
        $http.delete('todo/delete/'+deleteTodo).success(function(response){
            $scope.getAllTodos();
        }).error(function() {
            $scope.setError('Could not delete todo');
        });
    }

    $scope.deleteAllTodo = function(){
        $scope.resetError();
        $http.delete('todo/deleteAll').success(function(response){
            $scope.getAllTodos();
        }).error(function() {
            $scope.setError('Could not delete all todos');
        })
    }

    $scope.editTodo = function(position, todo){
        $scope.resetError();
        $scope.todoName = todo;
        $scope.position = position;
        $scope.editMode = true;
    }

    $scope.updateTodo = function(updateTodo){
        $scope.resetError();
        $http.put('todo/update/'+ $scope.position +'/'+updateTodo).success(function(response){
            $scope.getAllTodos();
            $scope.position = '';
            $scope.todoName = '';
            $scope.editMode = false;
        }).error(function(){
            $scope.setError('Could not update todo');
         })
    }

    $scope.resetTodoField = function() {
        $scope.resetError();
        $scope.todoName = '';
        $scope.editMode = false;
    };

    $scope.resetError = function() {
        $scope.error = false;
        $scope.errorMessage = '';
    };

    $scope.setError = function(message) {
        $scope.error = true;
        $scope.errorMessage = message;
    };

    $scope.getAllTodos();
}

File: 99-old-code/7.SpringMvc/in28minutes-services-springmvc/src/main/java/com/in28minutes/springmvc/services/web/controller/TodoController.java
package com.in28minutes.springmvc.services.web.controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseBody;

import com.in28minutes.springmvc.services.web.service.api.TodoService;

@Controller
@RequestMapping("/todo")
public class TodoController {

	@Autowired
	private TodoService todoService;

	@RequestMapping(value = "/all.json")
	public @ResponseBody List<String> viewAllTodos() {
		return todoService.allTodos();
	}

	@RequestMapping(value = "/add/{todo}", method = RequestMethod.POST)
	public @ResponseBody void addTodo(@PathVariable("todo") String todo) {
		todoService.addTodo(todo);
	}

	@RequestMapping(value = "/delete/{todo}", method = RequestMethod.DELETE)
	public @ResponseBody void deleteTodo(@PathVariable("todo") String todo) {
		todoService.deleteTodo(todo);
	}

	@RequestMapping(value = "/deleteAll", method = RequestMethod.DELETE)
	public @ResponseBody void deleteAllTodo() {
		todoService.deleteAll();
	}

	@RequestMapping(value = "/update/{position}/{todo}", method = RequestMethod.PUT)
	public @ResponseBody void updateTodo(
			@PathVariable("position") String position,
			@PathVariable("todo") String todo) {
		todoService.updateTodo(Integer.valueOf(position), todo);
	}

	@RequestMapping("/layout")
	public String getTodoPartialPage() {
		return "todo/layout";
	}
}
